cassovary-core/src/main/scala/com/twitter/cassovary/graph/Traverser.scala

```diff
@@ -396,7 +396,7 @@ trait QueueBasedTraverser extends Traverser {
 /**
  * A traverser that keeps track of first depth of visiting a given node.
  */
-trait DepthTracker extends QueueBasedTraverser {
+trait DepthTracker[NodeType <: Node] extends QueueBasedTraverser[NodeType] {
```

Comment:
Thought about this. I think 
```
trait DepthTracker  extends QueueBasedTraverser[Node]
```
is ok. Thanks to covariance QueueBasedTraverser[NodeType] <: QueueBasedTraverser[Node] so we can extend it with DepthTracker.